core/src/utils/prove.rs

```diff
@@ -240,8 +279,31 @@ where
     // Prove the shards.
     let mut deferred = ExecutionRecord::new(program.clone().into());
     let mut state = public_values.reset();
-    let mut shard_proofs = Vec::<ShardProof<SC>>::new();
     let mut report_aggregate = ExecutionReport::default();
+
+    let (tx, rx) = sync_channel::<Vec<ExecutionRecord>>(
+        opts.shard_batch_size * opts.shard_chunking_multiplier,
+    );
+
+    // Disable the reconstruct commitments in case it was enabled.
```

Comment:
i don't like how we override the ops here, can we just do commit_main, prove_shard directly like how we used to?